src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -104,12 +104,20 @@ QString BulkController::presetExtension() {
     return BULK_PRESET_EXTENSION;
 }
 
+void BulkController::visit(const KeyboardControllerPreset* preset) {
+    Q_UNUSED(preset);
+    qWarning() << "ERROR: Attempting to load a KeyboardControllerPreset to a BulkController!";
+    // TODO(XXX): throw a hissy fit.
+
+};
+
 void BulkController::visit(const MidiControllerPreset* preset) {
     Q_UNUSED(preset);
+    qWarning() << "ERROR: Attempting to load a MidiControllerPreset to a BulkController!";
     // TODO(XXX): throw a hissy fit.
-    qWarning() << "ERROR: Attempting to load a MidiControllerPreset to an HidController!";
 }
 
+// TODO Shouldn't this method raise a warning, just as visit(KeyboardControllerPreset*) and visit(MidiControllerPresetPreset*) ?
 void BulkController::visit(const HidControllerPreset* preset) {
```

Comment:
The BulkController uses HidControllerPresets as well, see https://github.com/mixxxdj/mixxx/blob/94e9c0c93d985119ffb6403d774882c02c4ce6a4/src/controllers/controllerpresetfilehandler.cpp#L46

However, you should consider to rename the class to  HidAndBulkControllerPreset to avoid future confusion.